### PR TITLE
feat: filter pins by status

### PR DESCRIPTION
**Pull request by GaryHomewood**

### CODE REVIEW

**Review by LeslieOA — APPROVED** [review]

File: packages/db/postgres/docker/docker-compose.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ version: '3.6'
 
 services:
   rest:
-    image: postgrest/postgrest:v8.0.0
```
Comment (flea89) [inline packages/db/postgres/docker/docker-compose.yml:5]: Is this required for the filtering to work?

If that's the case we need to make sure the version we're using on Heroku matches this one.

**Review by flea89 — DISMISSED** [review]
Great to see the progress on this one.
I've added a few comments here and there, let me know what you think.

File: packages/db/postgres/docker/docker-compose.yml (hunk 1 above)
Comment (GaryHomewood) [reply packages/db/postgres/docker/docker-compose.yml]: It is, and has already been bumped on main.

File: packages/db/index.js [hunk 2]
```diff
@@ -835,40 +863,58 @@ export class DBClient {
...
-      query = query.like('name', `${opts.name}`)
-    }
-
-    if (opts.name && match === 'iexact') {
-      query = query.ilike('name', `${opts.name}`)
-    }
+    if (!Object.keys(opts).length) {
+      query = query.eq('content.pins.status', PIN_STATUS[PIN_STATUS.Pinned])
+    } else {
+      if (opts.status) {
```
[the remark this answers is not in the document]
Comment (GaryHomewood) [reply packages/db/index.js]: Agree, the mapping of filter status to DB status should be the responsibility of the API

File: packages/db/db-client-types.ts [hunk 3]
```diff
@@ -294,7 +294,7 @@ export type ListPsaPinRequestOptions = {
   /**
    * status  to match
    */
-  status?: Array<definitions['pin']['status']>
+  statuses?: Array<definitions['pin']['status']>
```
Comment (GaryHomewood) [inline packages/db/db-client-types.ts:297]: This makes sense, given that the mapping of filter status to DB status is now in the API package. Although the type is now a mix of API filter attributes and DB attributes.

File: packages/db/index.js [hunk 4]
```diff
@@ -836,40 +862,45 @@ export class DBClient {
...
-
-    if (opts.name && match === 'exact') {
-      query = query.like('name', `${opts.name}`)
-    }
-
-    if (opts.name && match === 'iexact') {
-      query = query.ilike('name', `${opts.name}`)
-    }
+    if (!Object.keys(opts).length) {
+      query = query.eq('content.pins.status', 'Pinned')
```
Comment (GaryHomewood) [inline packages/db/index.js:870]: Should `Pinned` be some kind of enum?

Comment (flea89) [reply packages/db/index.js:870]: That's a really good point, I thought about it as well.
But given we already have statuses enums (that don't quite do for our case though) I'm afraid adding another one would be a bit confusing. 
So If it's ok with you I'd leave it out this specific ticket.

File: packages/db/db-client-types.ts (hunk 3 above)
Comment (flea89) [reply packages/db/db-client-types.ts:297]: I don't feel this represents a problem, the other options look reasonable in line.

**Review by vasco-santos — APPROVED** [review]
Good work!
lgtm

**Review by alanshaw — APPROVED** [review]
LGTM

**Pull request merged**